packages/vite/src/node/plugins/optimizedDeps.ts

```diff
@@ -77,9 +77,8 @@ export function optimizedDepsPlugin(): Plugin {
         try {
           return await fsp.readFile(file, 'utf-8')
         } catch {
-          const newMetadata = depsOptimizer.metadata
-          if (optimizedDepInfoFromFile(newMetadata, file)) {
-            // Outdated non-entry points (CHUNK), loaded after a rerun
+          if (browserHash) {
+            // Outdated optimized files loaded after a rerun
```

Comment:
This old condition did not match requests for stale optimized files.
The requests that was sent between "the optimizer starts the bundle" and "the optimizer finishes the bundle" is handled above after `await info.processing`, but the requests that was sent after "the optimizer finishes the bundle" doesn't match the condition above.

Assuming that cases like `new URL('./something', import.meta.url)` does not have a `?v=` query, I changed the condition to use that.